pymeasure/instruments/hp/hp3437A.py

```diff
@@ -0,0 +1,412 @@
+
+    @staticmethod
+    def _decode_trigger(self, status_bytes):
+        """Method to decode trigger mode
+
+        :param status_bytes: list of bytes to be decoded
+        :return trigger_mode: string with the current trigger mode
+        :rtype trigger_mode: str
+
+        """
+        cur_stat = self.Status(self.status_bytes(*status_bytes))
+        trig = cur_stat.b.trigger
+        if trig == 0:
+            log.error("HP3437A invalid trigger detected!")
+            trigger_mode = "INVALID"
+        if trig == 1:
+            trigger_mode = "Internal"
+        if trig == 2:
+            trigger_mode = "external"
+        if trig == 3:
+            trigger_mode = "hold/manual"
+        return trigger_mode
+
+    @staticmethod
+    def _unpack_data(self, data):
+        """
+        Method to unpack the data from the returned bytes in packed mode
+
+        :param data: list of bytes to be decoded
+        :return ret_data: float value
+
+        """
+        ret_data = self.packed_data(self.packed_bytes(*data))
+        return float(ret_data.b)
+
+    # commands overwriting the base implementaiton
+    def read(self):
+        """
+        Reads measured data from instrument, returns a np.array.
+
+        (This function also takes care of unpacking the data if required)
+
+        :return data: np.array containing the data
+        """
+        # Adjusting the timeout to match the number of counts and the delay
+
+        current_timeout = self.adapter.connection.timeout
+        time_needed = self.number_readings * self.delay
+        new_timeout = time_needed * 3 * 1000  # safety factor 3
+        # TODO Review required! (2021/12/29)
+        if new_timeout > current_timeout:
+            if new_timeout >= 1e6:
+                # Disables timeout if measurement would take more then 1000 sec
+                new_timeout = 1000000
+                log.info("HP3437A: timeout deactivated")
+            self.adapter.connection.timeout = new_timeout
+            log.info("HP3437A: timeout changed to %g", new_timeout)
+        read_data = self.adapter.connection.read_raw()
+        # check if data is in packed format format
+        if self.talk_ascii is not True:
+            processed_data = []
+            read_data_length = int(len(read_data) / 2)
+            for i in range(0, read_data_length):
+                _from = i * 2
+                _to = _from + 2
+                processed_data.append(self._unpack_data(self, read_data[_from:_to]))
+            self.adapter.connection.timeout = current_timeout
+            return np.array(processed_data)
+        self.adapter.connection.timeout = current_timeout
+        return np.array(read_data[:-2].decode("ASCII").split(","), dtype=float)
+
+    # commands/properties for instrument control
+    def check_errors(self):
+        """
+        As this instrument does not have a error indication bit,
+        this function alwyas returns 0.
+
+        """
+        return 0
```

Comment:
This method normally return a list, so would it be more appropriate to return `[]` ?

Reply:
> This method normally return a list, so would it be more appropriate to return `[]` ?
Yes good catch, will change this.